server/src/main/java/org/tctalent/server/api/system/SystemUploadApi.java

```diff
@@ -38,100 +37,100 @@
 @RequestMapping("api/system/upload")
 public class SystemUploadApi {
 
-    private static final Logger log = LoggerFactory.getLogger(SystemUploadApi.class);
+  private static final Logger log = LoggerFactory.getLogger(SystemUploadApi.class);
 
-    @Value("${aws.credentials.accessKey}")
-    private String accessKey;
+  @Value("${aws.credentials.accessKey}")
+  private String accessKey;
 
-    @Value("${aws.credentials.secretKey}")
-    private String secretKey;
+  @Value("${aws.credentials.secretKey}")
+  private String secretKey;
 
-    @Value("${aws.s3.bucketName}")
-    private String bucket;
+  @Value("${aws.s3.bucketName}")
+  private String bucket;
 
-    @Value("${aws.s3.upload-folder}")
-    private String uploadFolder;
+  @Value("${aws.s3.upload-folder}")
+  private String uploadFolder;
 
-    @Value("${aws.s3.max-size}")
-    private long maxSize;
+  @Value("${aws.s3.max-size}")
+  private long maxSize;
 
-    @GetMapping(value = "policy/{s3Key}")
-    public S3UploadData getUploadPolicy(@PathVariable("s3Key") String s3Key) throws Exception {
-        if (StringUtils.isNotBlank(uploadFolder)) {
-            s3Key = uploadFolder + "/" + s3Key;
-        }
-        return getPolicyForKey(s3Key);
+  @GetMapping(value = "policy/{s3Key}")
+  public S3UploadData getUploadPolicy(@PathVariable("s3Key") String s3Key) throws Exception {
+    if (StringUtils.isNotBlank(uploadFolder)) {
+      s3Key = uploadFolder + "/" + s3Key;
+    }
+    return getPolicyForKey(s3Key);
+  }
+
+  private S3UploadData getPolicyForKey(String s3Key) throws Exception {
+
+    Encoder encoder = Base64.getEncoder();
+
+    String policyDocument = "{\"expiration\": \"" + getExpirationTimestamp() + "Z\","
+        + " \"conditions\": [ "
+        + "      [\"starts-with\", \"$key\", \"" + s3Key + "\"],"
+        + "      {\"acl\": \"private\"},"
+        + "      {\"bucket\": \"" + bucket + "\"},"
+        + "      [\"starts-with\", \"$Content-Type\", \"\"],"
+        + "      [\"content-length-range\", 0, " + maxSize + "]"
+        + "   ]"
+        + "}";
+
+    String policy = encoder.encodeToString(policyDocument.getBytes(StandardCharsets.UTF_8))
+        .replaceAll("\n", "")
+        .replaceAll("\r", "");
+
+    Mac hmac = Mac.getInstance("HmacSHA1");
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This Message Authentication Code (MAC) uses a weak algorithm which might lead to cryptographic vulnerabilities. We recommend that you use the `SHA224`, `SHA256`, `SHA384`, `SHA512/224`, or `SHA512/256` algorithm.

[Learn more](https://cwe.mitre.org/data/definitions/327.html)